### ISSUE TITLE
wrong with nnUNetTrainer

### ISSUE THREAD

**Issue by frankhuaning** [MIC-DKFZ/nnUNet#1618]
Hi, i am freshman in medical segementation area. and i am facing some problem in nnUnetTrainer to chang my epochs. 
nnunetv2/training/nnUNetTrainer/variants/training_length/nnUNetTrainer_Xepochs.py
At this path, we can custom own epochs and train model with the given epochs. However, when i train model with 50, 100 or i inherit the function, it happens RuntimeError
` Could not find requested nnunet trainer nnUNetTrainer_50epochs in nnunetv2.training.nnUNetTrainer (/home/jupyter-wkh/.conda/envs/nnunet/lib/python3.10/site-packages/nnunetv2/training/nnUNetTrainer). If it is located somewhere else, please move it there.`
the code as below
`class nnUNetTrainer_50epochs(nnUNetTrainer):
    def __init__(self, plans: dict, configuration: str, fold: int, dataset_json: dict, unpack_dataset: bool = True,
                 device: torch.device = torch.device('cuda')):
        super().__init__(plans, configuration, fold, dataset_json, unpack_dataset, device)
        self.num_epochs = 50
class nnUNetTrainer_100epochs(nnUNetTrainer):
    def __init__(self, plans: dict, configuration: str, fold: int, dataset_json: dict, unpack_dataset: bool = True,
                 device: torch.device = torch.device('cuda')):
        super().__init__(plans, configuration, fold, dataset_json, unpack_dataset, device)
        self.num_epochs = 100
class nnUNetTrainer_500epochs(nnUNetTrainer):
    def __init__(self, plans: dict, configuration: str, fold: int, dataset_json: dict, unpack_dataset: bool = True,
                 device: torch.device = torch.device('cuda')):
        super().__init__(plans, configuration, fold, dataset_json, unpack_dataset, device)
        self.num_epochs = 500
`
by the way, what is the difference between nnUNetTrainer_Xepochs and nnUNetTrainer_Xepochs_NoMirrioring
wish all good!

**Comment by kabbas570** [comment]
@frankhuaning, you can easily change the num_epcoh variable to your desired variable of this file [nnUNetTrainer.py](https://github.com/MIC-DKFZ/nnUNet/blob/master/nnunetv2/training/nnUNetTrainer/nnUNetTrainer.py)

at line 145--         self.num_epochs = 1000,,, change 1000 to  your desired epochs.

Chees
Abbas

**Comment by frankhuaning** [comment]
@kabbas570 
Hi, kabbas. Thank for your answers. But unfortunately, i have tried in this way, this method is not work for me. Actually, i don't know where something wrong, because it seem everything good when i check `find_class_by_name.py`. 
Even the derived functions of nnUnet like (`nnUNetTrainer_50epochs`, `nnUNetTrainer_100epochs`) can't be used.

**Comment by kabbas570** [comment]
How you are accessing the variants of nnUNet, maybe there is some issue there then?

**Comment by frankhuaning** [comment]
sure, i have used the command like `nnUNetv2_train 14 2d 0 -tr nnUNetTrainer_50epochs --npz`
And, if i want to custom my epoch, its function wpuld be like
class nnUNetTrainer_500epochs(nnUNetTrainer):
 def __init__(self, plans: dict, configuration: str, fold: int, dataset_json: dict, unpack_dataset: bool = True, device: torch.device = torch.device('cuda')): 
super().__init__(plans, configuration, fold, dataset_json, unpack_dataset, device)
 self.num_epochs = 500
then,  `nnUNetv2_train 14 2d 0 -tr nnUNetTrainer_500epochs --npz`
thanks

**Comment by kabbas570** [comment]
@frankhuaning , Yes, I confirm that there is some bug in the latest version... I updated it a few days ago, and now I cannot control the num_epochs from [nnUNetTrainer.py](https://github.com/MIC-DKFZ/nnUNet/blob/master/nnunetv2/training/nnUNetTrainer/nnUNetTrainer.py)
@FabianIsensee  can you lease have a look, thanks

**Comment by sten2lu** [comment]
Hi @frankhuaning,

the difference between `nnUNetTrainer_Xepochs` and `nnUNetTrainer_Xepochs_NoMirrioring` is that there is no mirroring of the axis as data augmentation used during training for the second option whereas using mirrored axes is the default.

Otherwise, does the problem with the Trainer Classes which cannot be found still persisting for you?

**Comment by frankhuaning** [comment]
Hi @sten2lu 
Appreciate! now i understadn the difference between `nnUNetTrainer_Xepochs `and `nnUNetTrainer_Xepochs_NoMirrioring`.

Yes, I still have questions about the Trainer Classes, should i update the code?
Thanks!

**Comment by sten2lu** [comment]
Hi @frankhuaning,

Of course, you can ask them here.
For general questions, I would suggest reading the specific chapters of the [documentation](https://github.com/MIC-DKFZ/nnUNet/tree/master/documentation) first.

Just for my understanding, where would you update the code?

Best regards,
Carsten

**Comment by frankhuaning** [comment]
Hi @sten2lu 
Sorry, you may misunderstand my meaning. 
I do reading the specific chapters of the [documentation](https://github.com/MIC-DKFZ/nnUNet/tree/master/documentation) before, but I still have the problem with the Trainer Classes which cannot be found.

How can I fix it?

Best regards
Frank

**Comment by sten2lu** [comment]
Hi @frankhuaning,
did you update to the most recent version of nnUNetv2?

I just wanted to reproduce your error.
However, upon trying to run custom trainers myself to debug, everything ran smoothly. The described error appeared neither for the standard implemented nnUnetTrainer s.a. nnUNetTrainer_5epochs or for a custom implemented nnUNetTrainer.

Best regards,
Carsten

**Comment by frankhuaning** [comment]
Hi @sten2lu 
Appreciate! After I update to the most recent version of nnUNetv2, I fix the error and be able to custom my own nnUNetTrainer.
Thank you for your great job again!

cheers
Frank
